src/room/spawning/spawnFunctions.ts

```diff
@@ -35,7 +35,7 @@ Room.prototype.decideMaxCostPerCreep = function (maxCostPerCreep) {
     // If there are no sourceHarvesters or haulers
 
     if (
-        this.myCreeps.source1Harvester.length + this.myCreeps.source2Harvester.length === 0 ||
+        (this.myCreeps.source1Harvester?.length || 0) + (this.myCreeps.source2Harvester?.length || 0) === 0 ||
```

Comment:
I get what your doing but is there an change that one of then is zero if yes then keep it otherwise maybe only do for source2